sql/src/test/java/org/apache/druid/sql/calcite/expression/ExpressionsTest.java

```diff
@@ -270,6 +270,16 @@ public void testSubstring()
         makeExpression("substring(\"spacey\", (\"p\" - 1), \"p\")"),
         "hey"
     );
+    testHelper.testExpressionString(
```

Comment:
I've unpatched all non-test changes - and these cases still pass...did I miss something?

Reply:
I guess these tests here are only running on the native layer. And the issue which we face is more of coming in the calcite layer. Let me change the tests. Thanks for pointing these.

Moreover 
```select CONTAINS_STRING(null, 'a') -- gives INVALID use of null
select CONTAINS_STRING(dim3, 'a') from numFoo where dim3 is null -- fails the strict check```